src/mbgl/text/placement.cpp

```diff
@@ -534,11 +534,20 @@ void Placement::placeBucket(
 
 void Placement::commit(TimePoint now, const double zoom) {
     commitTime = now;
+    if (!getPrevPlacement()) {
```

Comment:
Is this only applicable for still & tile mode? Looks like in other cases, there is always valid previous placement provided by the controller. Maybe worth adding comment.

Reply:
will add an assertion